### PR TITLE
resolvectl: introduce dump-caches command

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/resolve/resolved-dns-cache.c [hunk 1]
```diff
@@ -1290,7 +1291,45 @@ int dns_cache_export_shared_to_packet(DnsCache *cache, DnsPacket *p) {
...
+
+        assert(a);
+        assert(*a);
+        assert(b);
+        assert(*b);
+
+        x = *a;
+        y = *b;
+        x_str = dns_resource_key_to_string(x->rr ? x->rr->key : x->key);
+        y_str = dns_resource_key_to_string(y->rr ? y->rr->key : y->key);
```
Comment (poettering) [inline src/resolve/resolved-dns-cache.c:1307]: dns_resource_key_compare_func() + dns_resource_record_compare_func() already exist

**Review by poettering — CHANGES_REQUESTED** [review]
I'd really prefer if we'd synthesize JSON from the cache instead, and return that to clients. It would be processable then, and shouldn't be too difficult to synthesize. i.e. add dns_resource_key_to_json() and dns_resource_key_from_json() or so, and similar for dns_resource_record(). Pass the RR RDATA stuff as base64 blob.

**Comment by keszybz** [comment]
Since @yuwata gave thumbsup for the json idea, let's mark this as "needs rework".

**Comment by poettering** [comment]
This reminds me of #19451 a bit, i.e. generate server-side json, pass to client via dbus, and then just prettify and output it there.

**Pull request closed without merging**